### PR TITLE
[Enhancement] Add tables_config in  information_schema db

### LINKED ISSUE #9498: Improve the Information_schema

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9498 

## Problem Summary(Required) ：

Add information_schema.tables_config to show tables config.
This table contains columns such as PRIMARY_KEY PARTITION_KEY
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 89 (19.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|6|00.00%|[67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|5|69|07.25%|[1332, 1333, 1335, 1336, 1338, 1339, 1340, 1341, 1343, 1344, 1345, 1346, 1347, 1348, 1349, 1351, 1353, 1354, 1356, 1359, 1360, 1361, 1371, 1373, 1374, 1375, 1377, 1380, 1381, 1383, 1384, 1385, 1387, 1388, 1389, 1390, 1391, 1392, 1393, 1394, 1395, 1396, 1397, 1398, 1400, 1401, 1402, 1403, 1404, 1406, 1407, 1408, 1410, 1411, 1412, 1413, 1417, 1418, 1419, 1420, 1421, 1422, 1423, 1424]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 91 (18.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|6|00.00%|[67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|5|71|07.04%|[1332, 1333, 1335, 1336, 1338, 1339, 1340, 1341, 1343, 1344, 1345, 1346, 1347, 1362, 1363, 1364, 1366, 1368, 1370, 1371, 1373, 1376, 1377, 1378, 1388, 1390, 1391, 1392, 1394, 1395, 1396, 1398, 1399, 1400, 1402, 1403, 1404, 1405, 1406, 1407, 1408, 1409, 1410, 1411, 1412, 1413, 1415, 1416, 1417, 1418, 1419, 1421, 1422, 1423, 1425, 1426, 1427, 1428, 1432, 1433, 1434, 1435, 1436, 1437, 1438, 1439]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 91 (18.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|6|00.00%|[67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|5|71|07.04%|[1332, 1333, 1335, 1336, 1338, 1339, 1340, 1341, 1343, 1344, 1345, 1346, 1347, 1349, 1350, 1351, 1356, 1362, 1365, 1366, 1368, 1371, 1372, 1373, 1383, 1385, 1386, 1387, 1389, 1390, 1391, 1393, 1394, 1395, 1397, 1398, 1399, 1400, 1402, 1403, 1404, 1405, 1406, 1407, 1408, 1409, 1411, 1412, 1413, 1414, 1415, 1417, 1418, 1419, 1421, 1422, 1423, 1424, 1428, 1429, 1430, 1431, 1432, 1433, 1434, 1435]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 87 / 91 (95.60%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|69|71|97.18%|[1400, 1402]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_be_unittest

File: be/src/exec/vectorized/schema_scanner/schema_tables_meta_scanner.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+SchemaScanner::ColumnDesc SchemaTablesMetaScanner::_s_table_tables_meta_columns[] = {
+        //   name,       type,          size,     is_null
+        {"TABLE_SCHEMA", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"TABLE_NAME", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"PRIMARY_KEY", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"PARTITION_KEY", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"DISTRIBUTE_KEY", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"DISTRIBUTE_TYPE", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"DISTRUBTE_BUCKET", TYPE_VARCHAR, sizeof(StringValue), false},
```
Comment (howrocks) [inline be/src/exec/vectorized/schema_scanner/schema_tables_meta_scanner.cpp:20]: DISTRIBUTE_BUCKET

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 2]
```diff
@@ -1317,4 +1326,112 @@ public TSetConfigResponse setConfig(TSetConfigRequest request) throws TException
...
+        int idx = 0;
+        try {
+            for (Column column : partitionInfo.getPartitionColumns()) {
+                if (idx != 0) {
+                    partitionKeySb.append(", ");
+                }
+                partitionKeySb.append("`").append(column.getName()).append("`");
+                idx++;
+            }
+        } catch (NotImplementedException e) {
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:1400]: NotImplementedException should be regarded as a bug, I suggest you check partitionInfo.getType() instead.

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 3]
```diff
@@ -1317,4 +1326,112 @@ public TSetConfigResponse setConfig(TSetConfigRequest request) throws TException
...
+        if (null != olapTable.getColocateGroup()) {
+            showProperties.put(PropertyAnalyzer.PROPERTIES_COLOCATE_WITH, olapTable.getColocateGroup());
+        }
+        Short replicationNum = olapTable.getDefaultReplicationNum();
+        showProperties.put(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM, String.valueOf(replicationNum));
+        tableMetaInfo.setProperties(showProperties.toString());
+        return tableMetaInfo;
+    }
+
+    private TTableMetaInfo genDefaultMetaInfo(TTableMetaInfo tableMetaInfo) {
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:1427]: "def" seems like a bug. I suggest you use "<unknown>" or "<not supported>".

File: fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        Database db = new Database(1, "test_db");
+        
+        List<Column> columns = new ArrayList<>();
+        columns.add(new Column("c1", Type.ARRAY_BOOLEAN));
+        columns.add(new Column("c2", Type.ARRAY_BOOLEAN));
+
+        // OlapTable
+        RangePartitionInfo partitionInfo = new RangePartitionInfo(columns);
+        HashDistributionInfo distributionInfo = new HashDistributionInfo(10, columns);
+        OlapTable table = new OlapTable(1, "test_table", columns, KeysType.PRIMARY_KEYS, partitionInfo, distributionInfo);
```
Comment (howrocks) [inline fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java:58]: You should test all kinds of tables, duplicate, aggregate, unique and primary key tables.

File: fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        TGetTablesMetaRequest req = new TGetTablesMetaRequest();
+        TGetTablesMetaResponse response = impl.getTablesMeta(req);
+        response.tables_meta_infos.forEach(info -> {
+            if (info.getTable_name().equals("test_table")) {
+                Assert.assertEquals("`c1`, `c2`", info.getPrimary_key());
+                Assert.assertEquals("`c1`, `c2`", info.getPartition_key());
+                Assert.assertEquals("10", info.getDistribute_bucket());
+                Assert.assertEquals("HASH", info.getDistribute_type());
+                Assert.assertEquals("`c1`, `c2`", info.getDistribute_key());
+                Assert.assertEquals("PRIMARY", info.getSort_key());
```
Comment (howrocks) [inline fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java:93]: why sort key is `PRIMARY`?

File: be/src/exec/vectorized/schema_scanner/schema_tables_meta_scanner.cpp (hunk 1 above)
Comment (waittttting) [reply be/src/exec/vectorized/schema_scanner/schema_tables_meta_scanner.cpp:20]: done

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java (hunk 3 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:1427]: Mysql is using “def”, so I think it's ok

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:1427]: ok then

File: fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java (hunk 5 above)
Comment (waittttting) [reply fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java]: done

File: fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java (hunk 4 above)
Comment (waittttting) [reply fe/fe-core/src/test/java/com/starrocks/service/FrontendServiceImplTest.java]: done

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 99 / 105 (94.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|81|85|95.29%|[1403, 1404, 1405, 1408]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java (hunk 2 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java]: done

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 99 / 105 (94.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|81|85|95.29%|[1403, 1404, 1405, 1408]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 99 / 105 (94.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|81|85|95.29%|[1403, 1404, 1405, 1408]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

File: be/src/exec/vectorized/schema_scanner/schema_helper.cpp [hunk 6]
```diff
@@ -78,6 +78,14 @@ Status SchemaHelper::get_table_privs(const std::string& ip, const int32_t port,
             [&request, &result](FrontendServiceConnection& client) { client->getTablePrivs(*result, request); });
 }
 
+Status SchemaHelper::get_tables_meta(const std::string& ip, const int32_t port, const TGetTablesMetaRequest& var_params,
```
Comment (howrocks) [inline be/src/exec/vectorized/schema_scanner/schema_helper.cpp:81]: tables_config is a better name for this table?  

information_schema.tables is also metadata.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 99 / 105 (94.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|81|85|95.29%|[1403, 1404, 1405, 1408]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_clang-tidy

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 99 / 105 (94.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|81|85|95.29%|[1403, 1404, 1405, 1408]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java [hunk 7]
```diff
@@ -459,6 +459,22 @@ public static Builder builder() {
...
+                                    "tables_config",
+                                    TableType.SCHEMA,
+                                    builder()
+                                            .column("TABLE_SCHEMA", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                            .column("TABLE_NAME", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                            .column("PRIMARY_KEY", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                            .column("PARTITION_KEY", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                            .column("DISTRIBUTE_KEY", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                            .column("DISTRIBUTE_TYPE", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                            .column("DISTRIBUTE_BUCKET", ScalarType.createVarchar(NAME_CHAR_LEN))
```
Comment (howrocks) [inline fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:474]: shoudl use integer type，not string？

File: be/src/exec/vectorized/schema_scanner/schema_helper.cpp (hunk 6 above)
Comment (waittttting) [reply be/src/exec/vectorized/schema_scanner/schema_helper.cpp]: done

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java (hunk 7 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 104 (94.23%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/SchemaTableType.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/catalog/DistributionInfo.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|80|84|95.24%|[1402, 1403, 1404, 1407]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|12|12|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_clang

**Comment by waittttting** [comment]
run starrocks_clang-tidy

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**